src/core/lib/security/security_connector/alts/alts_security_connector.cc

```diff
@@ -178,6 +178,15 @@ grpc_alts_auth_context_from_tsi_peer(const tsi_peer* peer) {
     gpr_log(GPR_ERROR, "Invalid or missing certificate type property.");
     return nullptr;
   }
+  /* Validate security level. */
+  const tsi_peer_property* security_level_prop =
+      tsi_peer_get_property_by_name(peer, TSI_SECURITY_LEVEL_PEER_PROPERTY);
+  if (security_level_prop == nullptr ||
```

Comment:
Why need to validate security level here? We just need to make sure it is set. Then we can faithfully plumb TSI value into here.